includes/WikiInitialize.php

```diff
@@ -355,22 +313,13 @@ public function loadExtensions() {
 
 			$list = array_column( $data['credits'], 'path', 'name' );
 
-			if ( $wgCreateWikiUsePhpCache ) {
-				$content = "<?php\n\n" .
-					"/**\n * Auto-generated extension list cache.\n */\n\n" .
-					'return ' . var_export( $list, true ) . ";\n";
-			} else {
-				$content = json_encode( $list );
-			}
+			$phpContent = "<?php\n\n" .
+				"/**\n * Auto-generated extension list cache.\n */\n\n" .
+				'return ' . var_export( $list, true ) . ";\n";
 
-			file_put_contents( "{$this->cacheDir}/{$extensionListFileName}", $content, LOCK_EX );
+			file_put_contents( "{$this->cacheDir}/extension-list.php", $phpContent, LOCK_EX );
```

Comment:
No because it is only created if it does *not* exist.

Reply:
What if its existence is checked while the file is being written?

Reply:
Then it still isn't going to cause a race condition because it will exist or won't its not going to write twice when we only write to it under one condition. At least as I assumed....